### PR TITLE
Allow customized Jenkins parameters for different pipelines and add aws integration deployment pipeline

### PR DESCRIPTION
**Pull request by shirangj**
Allow customized Jenkins parameters for different pipelines so that we can define different Jenkins parameters for a new pipeline and doesn't affect AR build parameters.

Add aws integration deployment pipeline.

### CODE REVIEW

File: scripts/build/Platform/Windows/pipeline.json
Comment (junbo75) [inline scripts/build/Platform/Windows/pipeline.json:20]: I think the batch script supports another parameter called COMMIT_ID, which is used to lock the version of CDK applications to deploy.

Comment (shirangj) [reply scripts/build/Platform/Windows/pipeline.json:20]: I will add COMMIT_ID as a new parameter too, thanks!

**Review by junbo75 — APPROVED** [review]
LG. I assume that we can update the description and value for those Jenkins parameters after this PR is merged?

**Comment by shirangj** [comment]
> LG. I assume that we can update the description and value for those Jenkins parameters after this PR is merged?

Yeah, I think it's better to leave it to you to update the description.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by brianherrera — APPROVED** [review]

**Comment by amzn-changml** [comment]
Shirang and I had an offline discussion about this, specifically around using the deployment script within the o3de `build_config.json` parameter. The main point is that this deployment pipeline needs to be chained to the awsi project test pipeline. 

Short term, this is the most frictionless way to do it, but long term, I'd like to see a way to separate out this pipeline with a minimum of code duplication (maybe with the chaining feature)

**Review by amzn-changml — APPROVED** [review]

**Pull request merged**
